features/verblijfstitelhistorie.feature

```diff
@@ -1,149 +1,333 @@
+    Als de verblijfstitelhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeildatum |
+    | burgerservicenummer | 999991553             |
+    | fields              | aanduiding.code, datumEinde, datumIngang |
+    | datumVan            | 2019-09-01            |
+    | datumTot            | 2019-06-30            |
+    Dan heeft de response een verblijfstitelhistorie met de waarden
+    | aanduiding.code  | datumIngang | datumEinde |
+    | 42               | 20160901    | 20190901   |
+
+  @gba
+  Scenario: Raadpleeg de verblijfstitelhistorie met periode binnen de geldigheidsperiode van een historische verblijfstitel
+    Als de verblijfstitelhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeildatum |
+    | burgerservicenummer | 999991553             |
+    | fields              | aanduiding.code, datumEinde, datumIngang |
+    | datumVan            | 2010-01-01            |
+    | datumTot            | 2010-12-31            |
+    Dan heeft de response een verblijfstitelhistorie met de waarden
+    | aanduiding.code  | datumIngang | datumEinde |
+    | 24               | 20090512    | 20120405   |
+
+  @gba
+  Scenario: Raadpleeg de verblijfstitelhistorie met periode over de geldigheidsperiodes van meerdere verblijfstitels
+    Als de verblijfstitelhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeildatum |
+    | burgerservicenummer | 999991553             |
+    | fields              | aanduiding.code, datumEinde, datumIngang |
+    | datumVan            | 2009-01-01            |
+    | datumTot            | 2009-06-30            |
+    Dan heeft de response een verblijfstitelhistorie met de waarden
+    | aanduiding.code  | datumIngang | datumEinde |
+    | 24               | 20090512    | 20120405   |
+    | 26               | 20081129    | 20090512   |
+
+  @gba
+  Scenario: Als voor een verblijfstitel ingangsdatum gelijk is aan datum einde, dan wordt deze verblijfstitel niet opgenomen in de response.
+    Als de verblijfstitelhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeildatum |
+    | burgerservicenummer | 999991553             |
+    | fields              | aanduiding.code, datumEinde, datumIngang |
+    | datumVan            | 2001-01-01            |
+    | datumTot            | 2002-01-02            |
+    Dan heeft de response geen verblijfstitelhistorie
+
+Rule: Als er meerdere al dan niet historische verblijfstitels elkaar in geldigheidsperiode overlappen dan wordt alleen de verblijfstitel met de m,eest recente datum opneming opgenomen in de response.
```

Comment:
Deze rule klopt denk ik niet zoals deze hier beschreven staat.  Ik denk dat het erom gaat dat er geen overlap kan zijn tussen verblijfstitels. Is deze rule niet hetzelfde als onderstaande rule "Er kan op enig moment in de tijd maximaal één verblijfstitel geldig zijn."
Daarbij kan alleen de proxy die overlap oplossen en moet de Gba variant wel beide leveren.

Een eerste stap zou m.i. moeten zijn dat een verblijfstitel met indicatie onjuist wordt genegeerd. De gba variant levert een onjuiste verblijfstitel niet.
```
    | categorie | Aanduiding | Datum einde | Ingangsdatum | Datum geldigheid | Onjuist |
    | 60        | 25         |             | 20091201     | 20091203         |         |
    | 60        | 26         | 20190901    | 20091129     | 20091203         |  O      |
```
In dit geval levert de Gba variant alleen de verblijfstitel met aanduiding 25 en begindatum 20091201.

Wanneer een verblijfstitel een begindatum heeft die op of voor de begindatum van de vorige verblijfstitel ligt, dan moet de eerdere verblijfstitel worden genegeerd. Deze filtering mag de gba-variant doen. Eigenlijk beschrijft dit een datafout, want hier had de vorige verblijfstitel indicatie onjuist moeten hebben.
```
    | categorie | Aanduiding | Datum einde | Ingangsdatum | Datum geldigheid | Onjuist |
    | 60        | 26         | 20101031    | 20091101     | 20091101         |         |
    | 60        | 26         | 20190901    | 20091129     | 20091203         |         |
```
In dit geval levert de gba-variant 1 verblijfstitel met ingangsdatum 20091101 en einddatum 20191031.

Wanneer de begindatum van de verblijfstitel eerder is dan de einddatum van de vorige, dan wordt de einddatum van de vorige gevuld met die begindatum. Hier kan alleen de proxy die overlap oplossen en moet de Gba variant wel beide ongewijzigd leveren.
```
    | categorie | Aanduiding | Datum einde | Ingangsdatum | Datum geldigheid | Onjuist |
    | 60        | 26         | 20210731    | 20200801     | 20200801         |         |
    | 60        | 26         | 20200831    | 20190901     | 20190902         |         |
```
Hier levert de gba-variant beide ongewijzigd en levert de proxy 2 verblijfstitels, waarvan de eerste begindatum 2020-08-01 en einddatum 2021-07-31 heeft en de tweede begindatum 2019-09-01 en einddatum 2020-08-01.